appinventor/blocklyeditor/src/versioning.js

```diff
@@ -1195,6 +1196,30 @@ Blockly.Versioning.tryReplaceBlockWithAssets = function(valueNode) {
   valueNode.appendChild(newBlock);
 }
 
+/**
+ * Replaces the block currently attached to the passed value input with a
+ * permissions dropdown block. The current block is replaced iff it is a text
+ * or number block.
+ * @param {Element} valueNode The node to modify.
+ */
+Blockly.Versioning.tryReplaceBlockWithPermissions =
+  function(valueNode, workspace) {
+    if (!valueNode) {
+      return;
+    }
+    var valueMap = Blockly.Versioning
+        .getOptionListValueMap(workspace, 'Permission');
```

Comment:
This comment isn't specific to this PR, but it seems like the option types are not fully qualified. This could have implications for extensions. Do you have a sense of the difficulty to make use of FQCNs for the options values rather than the simple name?

Reply:
Mr. Lyn and I discussed this, and we came to the conclusion that because the SCN is displayed on the block we wouldn't want to allow for OptionLists with conflicting SCNs. So rather than change the option values be to be based on the FQCN I'd rather change the ComponentProcessor to give a more explicit error if a developer creates a conflicting SCN.

That's just my opinion though :D

Reply:
It's an open world problem though, and so changing it in the component processor only detects collisions with extensions and core types. If internally we always use the FQCN (we can still present the simple name) then two extensions can have the same short name while internally maintaining two separate types. Generally, extension authors tend to prefix their extensions with some unique prefix, so this is likely a non-issue anyway.

Reply:
> It's an open world problem though, and so changing it in the component processor only detects collisions with extensions and core types.

Good point, I didn't consider different extensions having conflicts with eachother.

Afaik changing the dropdown blocks to be keyed by FQCN should be possible, but it would take some effort because I'd want to change the JSON representation to remove the "key" property from the dropdown data, and that change would have to flow through all of the other systems.

But if we do want to change it it is definitely better to do so before we release. Currently the SCNs are saved in the XML, and if you changed it to need the FQCN later there would be no way to cleanly upgrade extensions defining dropdown blocks. If we change it before release we can have the XML store the FQCN and everything will be consistent.

So do you want me to go ahead and look into this? Note it might be better to do so after the last few dropdown block PRs are in, that way I can fix up everything at once without worrying about rebasing changes.